digiroad2-geo/src/main/scala/fi/liikennevirasto/digiroad2/linearasset/NumericalLimitFiller.scala

```diff
@@ -171,10 +174,75 @@ object NumericalLimitFiller {
     }
   }
 
+  private def toSegment(persistedLinearAsset: PersistedLinearAsset) = {
+    (persistedLinearAsset.startMeasure, persistedLinearAsset.endMeasure)
+  }
+
+  /**
+    * Remove recursively all overlapping linear assets or adjust the measures if the overlap is smaller than the allowed tolerance.
+    * Keeping the order of the sorted sequence parameter.
+    * 1) Find a overlaped linear asset between the first linear asset of the sorted sequence and the tail
+    *   a) Split the overlaped linear asset and pick the linear assets minor than the allowed tolerence
+    *     If
+    *       the side code of the first linear asset and the overlaped linear asset are equal
+    *     OR
+    *       the first linear asset have both directions
+    *
+    * @param sorted Sorted sequence of Persisted Liner Assets
+    * @param result Recursive result of each iteration
+    * @return Sequence without overlapping linear assets
+    */
+  private def sortAndDrop(sorted: Seq[PersistedLinearAsset], result: Seq[PersistedLinearAsset]): Seq[PersistedLinearAsset] = {
+    val keeperOpt = sorted.headOption
+    if (keeperOpt.nonEmpty) {
+      val keeper = keeperOpt.get
+      val (overlapping) = sorted.tail.flatMap(asset => GeometryUtils.overlap(toSegment(keeper), toSegment(asset)) match {
+        case Some(overlap) =>
+          if (keeper.sideCode == asset.sideCode || keeper.sideCode == SideCode.BothDirections.value) {
+            Seq(
+              asset.copy(startMeasure = asset.startMeasure, endMeasure = overlap._1),
+              asset.copy(id = 0L, startMeasure = overlap._2, endMeasure = asset.endMeasure)
+            ).filter(a => a.endMeasure - a.startMeasure >= AllowedTolerance)
```

Comment:
After spliting the overlaped asset, if both are greater than AllowedTolerance they will be both filtered, even if it's not the ovelaped part. This is the expected result?